app/assets/javascripts/controllers/security_group/security_group_form_controller.js

```diff
@@ -1,54 +1,96 @@
-ManageIQ.angular.app.controller('securityGroupFormController', ['$scope', 'securityGroupFormId', 'miqService', 'API', function($scope, securityGroupFormId, miqService, API) {
-  $scope.securityGroupModel = { name: '' };
-  $scope.formId = securityGroupFormId;
-  $scope.afterGet = false;
-  $scope.modelCopy = angular.copy( $scope.securityGroupModel );
-  $scope.model = "securityGroupModel";
-
-  ManageIQ.angular.scope = $scope;
-
-  if (securityGroupFormId == 'new') {
-    $scope.securityGroupModel.name = "";
-    $scope.securityGroupModel.description = "";
-    $scope.newRecord = true;
-  } else {
-    miqService.sparkleOn();
-    API.get("/api/security_groups/" + securityGroupFormId + "?attributes=cloud_tenant").then(function(data) {
-      $scope.afterGet = true;
-      $scope.securityGroupModel.name = data.name;
-      $scope.securityGroupModel.description = data.description;
-      $scope.securityGroupModel.cloud_tenant_name = data.cloud_tenant.name;
-      $scope.modelCopy = angular.copy( $scope.securityGroupModel );
+ManageIQ.angular.app.controller('securityGroupFormController', ['securityGroupFormId', 'miqService', 'API', function(securityGroupFormId, miqService, API) {
+  var vm = this;
+
+  var init = function() {
+    vm.securityGroupModel = {
+      name: "",
+      description: "",
+      cloud_tenant_name: "",
+      firewall_rules: [],
+    };
+
+    getSecurityGroups();
+    vm.hostProtocols = ["", "TCP", "UDP", "ICMP"];
+    vm.networkProtocols = ["IPV4", "IPV6"];
+    vm.directions = ["inbound", "outbound"];
+
+    vm.formId = securityGroupFormId;
+    vm.model = "securityGroupModel";
+    vm.newRecord = securityGroupFormId === "new";
+    vm.saveable = miqService.saveable;
+
+    if (vm.newRecord) {
+      vm.afterGet = false;
+      vm.modelCopy = angular.copy( vm.securityGroupModel );
+    } else {
+      miqService.sparkleOn();
+      API.get("/api/security_groups/" + securityGroupFormId + "?attributes=name,description,cloud_tenant.name,firewall_rules").then(function(data) {
+        vm.securityGroupModel.name = data.name;
+        vm.securityGroupModel.description = data.description;
+        vm.securityGroupModel.cloud_tenant_name = angular.isDefined(data.cloud_tenant) ? data.cloud_tenant.name : undefined;
+        vm.securityGroupModel.firewall_rules = data.firewall_rules;
```

Comment:
Introduce a flag here that you can use down below in `vm.deleteFirewallRuleClicked`.
The idea is to set the form to dirty when a firewall rule is deleted using this flag, so that `Save` gets enabled.
```javascript
vm.securityGroupModel.firewall_rules_delete = false;
```